### DISCUSSION TITLE
pv backup issue

### DISCUSSION THREAD

**Discussion by vijaynetha** [vmware-tanzu/velero#4276] [General discussions]
Hi Team,

i deployed velero using snapshot way

command:- **velero install --provider aws --plugins registry.app.corpintra.net/velero/velero-plugin-for-aws:v1.2.1 --bucket <bucketname> --secret-file C:\Users\MAMKUMA\Desktop\Projects\velero\velero-v1.6.3-windows-amd64\credentials-velero --use-volume-snapshots=true --snapshot-location-config region=minio --backup-location-config region=minio,s3ForcePathStyle="true",s3Url=https://xx.xx.xx.xx:443**

when velero pod is up initial logs having below error. please help me on this issue.

**time="2021-10-25T08:27:49Z" level=error msg="Error checking repository for stale locks" controller=restic-repo error="error running command=restic unlock --repo=s3:https://s3-edc.emea.svc.corpintra.net:443/edcs3askbethabackupb/restic/database --password-file=/tmp/credentials/velero/velero-restic-credentials-repository-password --cache-dir=/scratch/.cache/restic, stdout=, stderr=Fatal: unable to open config file: Stat: The specified key does not exist.\nIs there a repository at the following location?\ns3:https://s3-edc.emea.svc.corpintra.net:443/edcs3askbethabackupb/restic/database\n: exit status 1" error.file="/go/src/github.com/vmware-tanzu/velero/pkg/restic/repository_manager.go:276" error.function="github.com/vmware-tanzu/velero/pkg/restic.(*repositoryManager).exec" logSource="pkg/controller/restic_repository_controller.go:144" name=database-default-2br4n namespace=velero**

**Comment by blackpiglet** [comment]
Please give more infomation about your environment. It seems restic reports some error, but the set-up command doesn't enable restic. 

The following are for your reference:

**What steps did you take and what happened:**
[A clear and concise description of what the bug is, and what commands you ran.)


**What did you expect to happen:**


**The output of the following commands will help us better understand what's going on**:
(Pasting long output into a [GitHub gist](https://gist.github.com) or other pastebin is fine.)

- `kubectl logs deployment/velero -n velero`
- `velero backup describe <backupname>` or `kubectl get backup/<backupname> -n velero -o yaml`
- `velero backup logs <backupname>`
- `velero restore describe <restorename>` or `kubectl get restore/<restorename> -n velero -o yaml`
- `velero restore logs <restorename>`


**Anything else you would like to add:**
[Miscellaneous information that will assist in solving the issue.]


**Environment:**

- Velero version (use `velero version`): 
- Velero features (use `velero client config get features`): 
- Kubernetes version (use `kubectl version`):
- Kubernetes installer & version:
- Cloud provider or hardware configuration:
- OS (e.g. from `/etc/os-release`):

**Comment by vijaynetha** [comment]
My kubernetes cluster is not supported to run restic daemonset. because of hostpath issue.
So iam running velero without restic.

iam trying to take postgress database backup.

****velero backup create database-5 --include-namespaces database
Backup request "database-5" submitted successfully.
Run `velero backup describe database-5` or `velero backup logs database-5` for more details.**

velero get backup
NAME         STATUS            ERRORS   WARNINGS   CREATED                         EXPIRES   STORAGE LOCATION   SELECTOR
database-5   PartiallyFailed   2        2          2021-10-25 14:34:38 +0530 IST   29d       default            <none>**

****>velero backup describe database-5
Name:         [1mdatabase-5[0m
Namespace:    velero
Labels:       velero.io/storage-location=default
Annotations:  velero.io/source-cluster-k8s-gitversion=v1.21.3-dhc
              velero.io/source-cluster-k8s-major-version=1
              velero.io/source-cluster-k8s-minor-version=21+

Phase:  [31mPartiallyFailed[0m (run `velero backup logs database-5` for more information)

Errors:    2
Warnings:  2

Namespaces:
  Included:  database
  Excluded:  <none>

Resources:
  Included:        *
  Excluded:        <none>
  Cluster-scoped:  auto

Label selector:  <none>

Storage Location:  default

Velero-Native Snapshot PVs:  auto

TTL:  720h0m0s

Hooks:  <none>

Backup Format Version:  1.1.0

Started:    2021-10-25 14:34:38 +0530 IST
Completed:  2021-10-25 14:34:41 +0530 IST

Expiration:  2021-11-24 14:34:38 +0530 IST

Total items to be backed up:  61
Items backed up:              61

Velero-Native Snapshots: <none included>****


After taking postgres database backup i can see error in velero pod logs 

time="2021-10-25T09:04:40Z" level=info msg="label \"topology.kubernetes.io/zone\" is not present on PersistentVolume, checking deprecated label..." backup=velero/database-5 logSource="pkg/backup/item_backupper.go:432" name=pvc-aff0f69c-83e9-40ed-9cb2-2676f8ea3c62 namespace= persistentVolume=pvc-aff0f69c-83e9-40ed-9cb2-2676f8ea3c62 resource=persistentvolumes
time="2021-10-25T09:04:40Z" level=info msg="label \"failure-domain.beta.kubernetes.io/zone\" is not present on PersistentVolume" backup=velero/database-5 logSource="pkg/backup/item_backupper.go:435" name=pvc-aff0f69c-83e9-40ed-9cb2-2676f8ea3c62 namespace= persistentVolume=pvc-aff0f69c-83e9-40ed-9cb2-2676f8ea3c62 resource=persistentvolumes
time="2021-10-25T09:04:40Z" level=error msg="Error getting volume snapshotter for volume snapshot location" backup=velero/database-5 error="rpc error: code = Unknown desc = config has invalid keys [s3ForcePathStyle s3Url]; valid keys are [region profile]" error.file="/go/pkg/mod/github.com/vmware-tanzu/velero@v1.6.2/pkg/plugin/framework/validation.go:50" error.function=github.com/vmware-tanzu/velero/pkg/plugin/framework.validateConfigKeys logSource="pkg/backup/item_backupper.go:449" name=pvc-aff0f69c-83e9-40ed-9cb2-2676f8ea3c62 namespace= persistentVolume=pvc-aff0f69c-83e9-40ed-9cb2-2676f8ea3c62 resource=persistentvolumes volumeSnapshotLocation=default


Environment
Velero version (use velero version):  **Version: v1.6.3**
Velero features (use velero client config get features):
Kubernetes version (use kubectl version): "v1.18.0
Kubernetes installer & version: **v1.21.3**
Cloud provider or hardware configuration: **openstack**
OS (e.g. from /etc/os-release):

**Comment by blackpiglet** [comment]
Hi,

From the information provided, the backup shows some error.

`Phase: �[31mPartiallyFailed�[0m (run velero backup logs database-5 for more information)
Errors: 2
Warnings: 2`

Please run `velero backup logs database-5` to find more trace to debug.

What is the storage location of PV backup? If restic is not used, the underpinned cloud service should provide some snapshot mechanism to support the volume backup.
